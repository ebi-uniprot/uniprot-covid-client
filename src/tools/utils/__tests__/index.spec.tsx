import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';

import { getServerErrorDescription, getJobMessage } from '..';

import createdJob from '../../__mocks__/internal-jobs/created';
import runningJob from '../../__mocks__/internal-jobs/running';

describe('getServerErrorDescription', () => {
  it('should get formatted error string from server error', () => {
    const data = `<?xml version='1.0' encoding='UTF-8'?>
    <error>
      <description>Invalid parameters: 
        Sequence -> Error in reading input sequence. Please check your input.</description>
    </error>`;
    const error = { response: { data } };
    expect(getServerErrorDescription(error)).toEqual(`Invalid parameters: 
        Sequence â†’ Error in reading input sequence. Please check your input.`);
  });
});

describe('getJobMessage', () => {
  it('should create an error message object', () => {
    expect(
      getJobMessage({
        job: createdJob,
        errorDescription: 'Some error',
      })
    ).toEqual({
      id: 'local-97e5ab00-9ff0-11ea-baf5-bf14c0760612',
      format: 'POP_UP',
      tag: 'JOB',
      omitAndDeleteAtLocations: ['Dashboard'],
      content: 'Some error',
      level: 'failure',
    });
  });

  it('should create a success message object', () => {
    const jobMessage = getJobMessage({
      job: runningJob,
      nHits: 100,
    });
    const { asFragment } = render(
      <Router>{jobMessage.content as JSX.Element}</Router>
    );
    expect(asFragment()).toMatchSnapshot();
    delete jobMessage.content;
    expect(jobMessage).toEqual({
      id: 'local-97e5ab00-9ff0-11ea-baf5-bf14c9060612',
      format: 'POP_UP',
      tag: 'JOB',
      omitAndDeleteAtLocations: ['Dashboard'],
      level: 'success',
    });
  });
});
