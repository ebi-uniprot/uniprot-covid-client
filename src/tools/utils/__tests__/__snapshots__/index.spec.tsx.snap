// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getJobMessage should create a success message object 1`] = `
<DocumentFragment>
  blast job 
  <a
    href="/blast/ncbiblast-R20200522-953245-6299-98843150-p1m/overview"
  >
    "my job title"
  </a>
   finished, found 100 hits
</DocumentFragment>
`;
